# Práctica Clase 1 

1) Crear una función capaz de convertir números enteros de base 10 a base 2. Debe recibir como parámetro el número a convertir<br>
Consideraciones:<br> 
a. Tratar de resolverlo sin usar la función format(nro,"b")<br>
b. El pdf "conversion-de-decimal-a-binario.pdf" puede servir de ayuda.


In [104]:
def NumeroBinario(numero):
    '''
    Esta función recibe como parámetro un número entero mayor ó igual a cero y lo devuelve en su 
    representación binaria. Debe recibir y devolver un valor de tipo entero.
    En caso de que el parámetro no sea de tipo entero y mayor a -1 retorna nulo.
    '''
    if type(numero)!=int:
        return None
    if numero<0:
        return None
    if numero==0:
        return 0
    r=''
    while numero>0:
        r=str(numero%2)+r
        numero//=2
    return int(r)


In [188]:
format(4, 'b')

'100'

In [189]:
#Ejemplo:
NumeroBinario(4)


100

In [180]:
def fraccion_binario(n):
    op=n
    bin='0.'
    while op!=1:
        if len(bin)>26:
            break
        op=op*2
        if op<1:
            bin=bin+'0'
        else:
            bin=bin+'1'
        if op>1:
            op-=1
    return float(bin)


In [181]:
frac=[1/2, 1/3, 1/4, 1/5, 1/6, 1/7, 1/8, 1/9]

In [184]:
for a in frac:
    print('fraccion', a,'numero binario', fraccion_binario(a))

fraccion 0.5 numero binario 0.1
fraccion 0.3333333333333333 numero binario 0.010101010101010102
fraccion 0.25 numero binario 0.01
fraccion 0.2 numero binario 0.0011001100110011
fraccion 0.16666666666666666 numero binario 0.00101010101010101
fraccion 0.14285714285714285 numero binario 0.001001001001001001
fraccion 0.125 numero binario 0.001
fraccion 0.1111111111111111 numero binario 0.000111000111000111


2) Convertir de decimal a binario las fracciones 1/2, 1/3, 1/4, 1/5, 1/6, 1/7, 1/8, 1/9. Luego analizar los resultados y observar qué particularidad se encuentra en los mismos. Se puede usar Python o una calculadora, lo importante es ver si hay algo que podemos notar...

Salida esperada:
* Fraccion 1 / 2 :  0.5  y En binario:  0.1
* Fraccion 1 / 3 :  0.3333333333333333  y En binario:  0.010101010101010101010101
* Fraccion 1 / 4 :  0.25  y En binario:  0.01
* Fraccion 1 / 5 :  0.2  y En binario:  0.001100110011001100110011
* Fraccion 1 / 6 :  0.16666666666666666  y En binario:  0.001010101010101010101010
* Fraccion 1 / 7 :  0.14285714285714285  y En binario:  0.001001001001001001001001
* Fraccion 1 / 8 :  0.125  y En binario:  0.001
* Fraccion 1 / 9 :  0.1111111111111111  y En binario:  0.000111000111000111000111